.github/workflows/nix-compile.yml

```
@@ -0,0 +1,265 @@
+name: "Nix pipeline"
```

Let's not name stuff based on the tooling, but instead based on what it does.

```suggestion
name: "Build"
```